app/controllers/concerns/foreman_puppet/api/import_puppetclasses_common_controller.rb

```diff
@@ -74,7 +74,13 @@ def changed_environments
         end
 
         unless @environments.any?
-          render_message(_('No changes to your environments detected'))
+          # check if environment id/name was valid at all
+          if @env_id && (Environment.authorized(:view_environments).find_by(id: @env_id) || \
+             Environment.authorized(:view_environments).find_by(name: @env_id))
+            render_message(_('No changes to your environments detected'))
+          else
+            render_message(_('The environment with id %s cannot be found.') % @env_id)
```

Comment:
The fact, that it is not in DB, should not be a failure, we can be importing a new environment.
If it doesn't exists, you might be just trying to import new env, that you've misspelled.

Wouldn't it be easier just detect whether `env_id` is set and if it is let the message be something like:
`No changes for environment <env_id> detected`?

Reply:
> The fact, that it is not in DB, should not be a failure, we can be importing a new environment.

This is checked earlier. If a new environment is detected before, it will skip that code.  

> what about when we do not pass the env_id

You are right, I need to adjust the message if there is no `@env_id`.

> Wouldn't it be easier just detect whether env_id is set and if it is let the message be something like:

Easier, I agree. Just not sure if the user realizes that he/she misspelled the environment. IMHO it might be clearer to explicitly tell that it is not existing.